### PR TITLE
Make account info mandatory for AuthenticationResult and CacheRecord types.

### LINKED ISSUE #2882: When can AccountInfo be null after login resolves?
## Library

- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [x] `@azure/msal-browser@2.9.0`
- [ ] `@azure/msal-node@1.x.x`
- [ ] `@azure/msal-react@1.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angular@2.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Description

I'm upgrading from 2.4.1 to 2.9.0 and I'm running into new type errors when accessing `AccountInfo` on the resolved value of the login APIs, example:
```ts
const ssoResult = await msal.ssoSilent(...);
console.log(ssoResult.account.username); // Object is possibly 'null'. ts(2531)
```

I'd like to know if there is now a potential scenario where the login api's can resolve but not include `AccountInfo` data? If so, any way to avoid this scenario?

## Source

- [x] Internal (Microsoft)
- [ ] Customer request

### PR DESCRIPTION
**Pull request by konstantin-msft**
- Make account info mandatory for `AuthenticationResult` type.
- Make account info mandatory for `CacheRecord` type.

FYI: 1p will be updated thereafter.

### CODE REVIEW

File: lib/msal-browser/test/app/PublicClientApplication.spec.ts [hunk 1]
```diff
@@ -94,6 +93,7 @@ import { NativeTokenRequest } from "../../src/broker/nativeBroker/NativeRequest"
 import { NativeAuthError } from "../../src/error/NativeAuthError";
 import { StandardController } from "../../src/controllers/StandardController";
 import { BrowserPerformanceMeasurement } from "../../src/telemetry/BrowserPerformanceMeasurement";
+import { AuthenticationResult } from "../../src";
```
Comment (tnorling) [inline lib/msal-browser/test/app/PublicClientApplication.spec.ts]:
```suggestion
import { AuthenticationResult } from "../../src/response/AuthenticationResult";
```

File: lib/msal-browser/test/interaction_client/PopupClient.spec.ts [hunk 2]
```diff
@@ -56,6 +55,7 @@ import {
 import { FetchClient } from "../../src/network/FetchClient";
 import { InteractionHandler } from "../../src/interaction_handler/InteractionHandler";
 import { getDefaultPerformanceClient } from "../utils/TelemetryUtils";
+import { AuthenticationResult } from "../../src";
```
Comment (tnorling) [inline lib/msal-browser/test/interaction_client/PopupClient.spec.ts]:
```suggestion
import { AuthenticationResult } from "../../src/response/AuthenticationResult";
```

File: lib/msal-browser/test/interaction_client/RedirectClient.spec.ts [hunk 3]
```diff
@@ -72,6 +71,7 @@ import { EventType } from "../../src/event/EventType";
 import { NativeInteractionClient } from "../../src/interaction_client/NativeInteractionClient";
 import { NativeMessageHandler } from "../../src/broker/nativeBroker/NativeMessageHandler";
 import { getDefaultPerformanceClient } from "../utils/TelemetryUtils";
+import { AuthenticationResult } from "../../src";
```
Comment (tnorling) [inline lib/msal-browser/test/interaction_client/RedirectClient.spec.ts]:
```suggestion
import { AuthenticationResult } from "../../src/response/AuthenticationResult";
```

**Review by tnorling — APPROVED** [review]
A couple nits but looks good! I believe there's also an open issue for this, please link it if it's still open, thanks!

**Review by peterzenz — APPROVED** [review]

**Pull request merged**